src/sage/schemes/curves/affine_curve.py

```diff
@@ -314,52 +314,132 @@ def _repr_type(self):
+                sage: P2 = AffineSpace(2, F, names='xy')
+                sage: R = P2.coordinate_ring()
+                sage: x, y = R.gens()
+                sage: f = y^2 - x^9 - x
+                sage: C = Curve(f)
+                sage: K = FractionField(R)
+                sage: r = 1/x
+                sage: C.divisor_of_function(r)
+                [(-2, (0, 0))]
+                sage: r = x^3
+                sage: C.divisor_of_function(r)
+                [(6, (0, 0))]
+        """
+        
+        def intersection_number(P, Q, pt=(0,0)):
+            """
+                Return the multiplicity of intersection of P and Q at pt.
+            
+                INPUT: P and Q are polynomials in 2 variables
+            
+                OUTPUT:
+            
+            
+                -  ``int`` - The multiplicity of intersection of P and Q at pt
+            
+            
+                EXAMPLES::
+            
+                    sage: F = GF(5)
+                    sage: P2 = AffineSpace(2, F, names='xy')
+                    sage: R = P2.coordinate_ring()
+                    sage: x, y = R.gens()
+                    sage: f = y^2 - x^9 - x
+                    sage: intersection_number(f, x)
+                    2
+                    sage: intersection_number(f, x^3)
+                    6
+            """
+            if not(P.parent() == Q.parent()):
+                raise ValueError("P and Q must be in the same ambiant space")
+            x, y = P.parent().gens()
+            P = P(x + pt[0], y + pt[1]); Q = Q(x + pt[0], y + pt[1])
+            if P(0,0)!=0 or Q(0,0)!=0:
+                return 0
+            p=P(x,0);r=p.degree(x)
+            q=Q(x,0);s=q.degree(x)
+            if s < r :
+                P,Q,p,q,r,s = Q,P,q,p,s,r
+            if P==0:
+                return Infinity
+            if r==-1:
+                P=P.quo_rem(y)[0]
+                v=q.univariate_polynomial().valuation()
+                return intersection_number(P, Q) + v
+            a=p.lc();b=q.lc()
+            Q=a*Q-b*x^(s-r)*P
+            return intersection_number(P, Q)
+        
+        P = self.defining_polynomial()
+        x, y = P.parent().gens()
+        P2 = AffineSpace(2, F, names='xy')
+        In = []
+        if r.numerator() == 1:
+            pass
+        else :
+            Q = P.parent()(r.numerator())
+            H = gcd(P,Q)
+            if H != 1:
+                raise ValueError("r=0")
+            elif Q == 1:
+                pass
+            else:
+                Rx = P.resultant(Q,x)
+                Or = Rx.univariate_polynomial().roots(multiplicities=false)
+                Ry = P.resultant(Q,y)
+                Ab = Ry.univariate_polynomial().roots(multiplicities=false)
+                for a in Ab:
+                    for b in Or:
+                        if P(a,b) == 0 and Q(a,b) == 0:
+                            In = In + [(intersection_number(P,Q,(a,b)),P2(a,b))]
```

Comment:
```suggestion
                            In.append((P2(a, b), intersection_number(P, Q, (a, b))))
```
For your information, `+ [...]` is very slow, `.append` can be used for appending a single element to the end of a list, or `.extend` for multiple.

Reply:
I swap the elements for the change below